infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/configurator/GitconfigUserDataConfigurator.java

```diff
@@ -84,13 +84,13 @@ public void configure(NamespaceResolutionContext namespaceResolutionContext, Str
             .getItems()
             .stream()
             .noneMatch(
-                configMap ->
-                    configMap
-                            .getMetadata()
-                            .getAnnotations()
-                            .entrySet()
-                            .containsAll(annotations.entrySet())
-                        && configMap.getData().containsKey(CONFIGMAP_DATA_KEY))) {
+                configMap -> {
+                  Map<String, String> annotationsMap = configMap.getMetadata().getAnnotations();
```

Comment:
metadata can not be null, right?

Reply:
It can't, it is a mandatory field and contains secret name.